src/main/java/edu/cornell/kfs/fp/batch/service/impl/AccountingXmlDocumentDownloadAttachmentServiceImpl.java

```diff
@@ -34,34 +35,37 @@
     protected WebServiceCredentialService webServiceCredentialService;
 
     @Override
-    public Attachment createAttachmentFromBackupLink(Document document,
-            AccountingXmlDocumentBackupLink accountingXmlDocumentBackupLink) throws IOException {
+    public Attachment createAttachmentFromBackupLink(Document document, AccountingXmlDocumentBackupLink accountingXmlDocumentBackupLink) {
         if (StringUtils.isBlank(accountingXmlDocumentBackupLink.getCredentialGroupCode())) {
             LOG.error("createAttachmentFromBackupLink, the Credential Group Code is blank");
-            throw new IOException("Unable to download attachment with blank Credential Group Code: " + accountingXmlDocumentBackupLink.getLinkUrl());
+            throw new ValidationException("Unable to download attachment with blank Credential Group Code: " + accountingXmlDocumentBackupLink.getLinkUrl());
         }
-        
-        byte[] formFile = downloadByteArray(accountingXmlDocumentBackupLink);
-        
-        if (formFile.length > 0) {
-            String uploadFileName = accountingXmlDocumentBackupLink.getFileName();
-            String mimeType = URLConnection.guessContentTypeFromName(uploadFileName);
-            int fileSize = (int) formFile.length;
-            String attachmentType = null;
-            
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("createAttachmentFromBackupLink, uploadFileName: " + uploadFileName + " mimeType: " + mimeType
-                    + " fileSize: " + fileSize);
+        try {
+            byte[] formFile = downloadByteArray(accountingXmlDocumentBackupLink);
+
+            if (formFile.length > 0) {
+                String uploadFileName = accountingXmlDocumentBackupLink.getFileName();
+                String mimeType = URLConnection.guessContentTypeFromName(uploadFileName);
+                int fileSize = (int) formFile.length;
+                String attachmentType = null;
+
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("createAttachmentFromBackupLink, uploadFileName: " + uploadFileName + " mimeType: " + mimeType
+                            + " fileSize: " + fileSize);
+                }
+                InputStream inputStream = new ByteArrayInputStream(formFile);
+
+                Attachment attachment = attachmentService.createAttachment(document, uploadFileName, mimeType, fileSize, inputStream, attachmentType);
+                return attachment;
+
+            } else {
+                LOG.error("createAttachmentFromBackupLink, the form file is NULL");
+                throw new ValidationException("The form file is NULL: " + accountingXmlDocumentBackupLink.getLinkUrl());
```

Comment:
Did you see Jay's comment on this line? (I received an email for it, but for some reason I don't see the comment in the GitHub UI.) Jay thought that the generic cannot-download message would be better for this spot, given that this is something users will see in the report. At the very least, should this be changed to mention something other than "form file" if that would confuse the users?

Reply:
hmm, I did not see it but good point. Would this work better: 
throw new ValidationException("Unable to download attachment: " + accountingXmlDocumentBackupLink.getLinkUrl());

Reply:
Yes, I believe a message like that would work better in this spot.

Reply:
I have updated the code. I think the logged message should stay in it's current form. thank you.

Reply:
:-)